Crash with Zappr integration

Just had a crash when trying to create a repo due to the Zappr client:

```
➜ github-cli repo create my-repo-name                                                                                                                                                                        (kubeconfig is not found)
• Creating repository...   
• Adding labels to repository...
• Adding Zappr...          
• Adding teams to repository...
• Adding branch protections to repository...
• Repository created! 
 Here is how to access it git://github.com/hellofresh/my-repo-name.git
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x1524bef]

goroutine 82 [running]:
github.com/hellofresh/github-cli/pkg/zappr.(*clientImpl).doRequest(0xc4201ca270, 0xc42011e000, 0x0, 0x0, 0x1, 0xc4205e43f0)
	/go/src/github.com/hellofresh/github-cli/pkg/zappr/client.go:147 +0x15f
github.com/hellofresh/github-cli/pkg/zappr.(*clientImpl).Enable(0xc4201ca270, 0xaeb3fdf, 0xc4201a49c0, 0x19a)
	/go/src/github.com/hellofresh/github-cli/pkg/zappr/client.go:83 +0x101
github.com/hellofresh/github-cli/cmd.RunCreateRepo.func2(0x8, 0x1772cb8)
	/go/src/github.com/hellofresh/github-cli/cmd/repo_create.go:159 +0x1e5
github.com/hellofresh/github-cli/vendor/golang.org/x/sync/errgroup.(*Group).Go.func1(0xc42001dbc0, 0xc4205b41e0)
	/go/src/github.com/hellofresh/github-cli/vendor/golang.org/x/sync/errgroup/errgroup.go:58 +0x57
created by github.com/hellofresh/github-cli/vendor/golang.org/x/sync/errgroup.(*Group).Go
	/go/src/github.com/hellofresh/github-cli/vendor/golang.org/x/sync/errgroup/errgroup.go:55 +0x66
```

Also, integration with Zappr didn't happen. No webhooks and no branch protection rules.

--- a/pkg/zappr/client.go
+++ b/pkg/zappr/client.go
@@ -145,7 +145,9 @@ func (c *clientImpl) doRequest(req *http.Request) (int, *zapprErrorResponse, err
 	resp, err := c.slingClient.Do(req, nil, zapprErrorResponse)
 
 	if resp == nil {
-		return http.StatusInternalServerError, nil, errwrap.Wrap(ErrZapprServerError, err)
+		// Even though 0 does not seem like a valid http response code
+		// The status would be ignored by any calling code anyway since a non-nil error is returned
+		return 0, nil, errwrap.Wrap(ErrZapprServerError, err)
 	}
 
 	if resp.StatusCode == http.StatusUnauthorized {

--- a/pkg/zappr/client.go
+++ b/pkg/zappr/client.go
@@ -145,7 +145,7 @@ func (c *clientImpl) doRequest(req *http.Request) (int, *zapprErrorResponse, err
 	resp, err := c.slingClient.Do(req, nil, zapprErrorResponse)
 
 	if resp == nil {
-		return http.StatusInternalServerError, nil, ErrZapprServerError
+		return http.StatusInternalServerError, nil, errwrap.Wrap(ErrZapprServerError, err)
 	}
 
 	if resp.StatusCode == http.StatusUnauthorized {

--- a/pkg/zappr/client.go
+++ b/pkg/zappr/client.go
@@ -144,6 +144,10 @@ func (c *clientImpl) doRequest(req *http.Request) (int, *zapprErrorResponse, err
 	zapprErrorResponse := &zapprErrorResponse{}
 	resp, err := c.slingClient.Do(req, nil, zapprErrorResponse)
 
+	if resp == nil {
+		return http.StatusInternalServerError, nil, ErrZapprServerError
+	}
+
 	if resp.StatusCode == http.StatusUnauthorized {
 		return resp.StatusCode, nil, ErrZapprUnauthorized
 	}
@@ -153,7 +157,7 @@ func (c *clientImpl) doRequest(req *http.Request) (int, *zapprErrorResponse, err
 			return resp.StatusCode, zapprErrorResponse, errwrap.Wrap(errors.New(zapprErrorResponse.Detail), ErrZapprServerError)
 		}
 
-		return resp.StatusCode, nil, ErrZapprServerError
+		return resp.StatusCode, nil, errwrap.Wrap(ErrZapprServerError, err)
 	}
 
 	return resp.StatusCode, nil, err

--- a/pkg/zappr/client_test.go
+++ b/pkg/zappr/client_test.go
@@ -275,6 +275,23 @@ func TestDisable_RepoDeletedFromGithub_AndNotOnZappr(t *testing.T) {
 	// Assert "already not enabled" error was returned
 	assert.Equal(t, ErrZapprAlreadyNotEnabled, err)
 
-	// Assert expected calls we made to Zappr
+	// Assert expected calls were made to Zappr
+	zapprMock.AssertExpectations(t)
+}
+
+func TestProblematicRequest(t *testing.T) {
+	// Get the Zappr Client, Mock Handler and Test Server
+	client, zapprMock, testServer := NewMockAndHandlerNilResponse()
+
+	// Start the test server and stop it when done
+	testServer.Start()
+	defer testServer.Close()
+
+	err := client.Enable(1)
+
+	// Assert "unknown zappr" error was returned
+	assert.True(t, errwrap.Contains(err, ErrZapprServerError.Error()))
+
+	// Assert expected calls were made to Zappr
 	zapprMock.AssertExpectations(t)
 }

--- a/pkg/zappr/client_test.go
+++ b/pkg/zappr/client_test.go
@@ -32,7 +32,7 @@ func TestAuthWithGithubToken(t *testing.T) {
 
 	client.Enable(1)
 
-	// Assert expected calls we made to Zappr
+	// Assert expected calls were made to Zappr
 	zapprMock.AssertExpectations(t)
 }
 
@@ -58,7 +58,7 @@ func TestAuthWithZapprToken(t *testing.T) {
 
 	client.Enable(1)
 
-	// Assert expected calls we made to Zappr
+	// Assert expected calls were made to Zappr
 	zapprMock.AssertExpectations(t)
 }
 
@@ -94,7 +94,7 @@ func TestArbitraryErrorWithJSONErrorBodyFromZappr(t *testing.T) {
 	// Assert error detail retured by zappr is in returned error
 	assert.True(t, errwrap.Contains(err, "Strange error message"), err)
 
-	// Assert expected calls we made to Zappr
+	// Assert expected calls were made to Zappr
 	zapprMock.AssertExpectations(t)
 }
 
@@ -122,7 +122,7 @@ func TestArbitraryErrorWithNonJSONErrorBodyFromZappr(t *testing.T) {
 	// Assert "unknown zappr" error was returned
 	assert.True(t, errwrap.Contains(err, ErrZapprServerError.Error()))
 
-	// Assert expected calls we made to Zappr
+	// Assert expected calls were made to Zappr
 	zapprMock.AssertExpectations(t)
 }
 
@@ -149,7 +149,7 @@ func TestEnable(t *testing.T) {
 	// Assert no errors were received
 	assert.Nil(t, err)
 
-	// Assert expected calls we made to Zappr
+	// Assert expected calls were made to Zappr
 	zapprMock.AssertExpectations(t)
 }
 
@@ -182,7 +182,7 @@ func TestEnable_AlreadyExist(t *testing.T) {
 	// Assert "already enabled" error was returned
 	assert.Equal(t, ErrZapprAlreadyEnabled, err)
 
-	// Assert expected calls we made to Zappr
+	// Assert expected calls were made to Zappr
 	zapprMock.AssertExpectations(t)
 }
 
@@ -209,7 +209,7 @@ func TestDisable(t *testing.T) {
 	// Assert no errors were received
 	assert.Nil(t, err)
 
-	// Assert expected calls we made to Zappr
+	// Assert expected calls were made to Zappr
 	zapprMock.AssertExpectations(t)
 }
 
@@ -242,7 +242,7 @@ func TestDisable_RepoDeletedFromGithub(t *testing.T) {
 	// Assert "already not enabled" error was returned
 	assert.Equal(t, ErrZapprAlreadyNotEnabled, err)
 
-	// Assert expected calls we made to Zappr
+	// Assert expected calls were made to Zappr
 	zapprMock.AssertExpectations(t)
 }
 

--- a/pkg/zappr/zappr_mock.go
+++ b/pkg/zappr/zappr_mock.go
@@ -1,6 +1,7 @@
 package zappr
 
 import (
+	"errors"
 	"fmt"
 	"net/http"
 	"net/url"
@@ -46,11 +47,31 @@ func newMockAndHandler() (*http.Client, *test.MockHandler, *test.Server) {
 	return httpClient, mockHandler, mockServer
 }
 
+func newMockAndHandlerWithNilResponseTransport() (*http.Client, *test.MockHandler, *test.Server) {
+	mockHandler := &test.MockHandler{}
+	mockServer := test.NewUnstartedServer(mockHandler)
+
+	httpClient := &http.Client{
+		Transport: &NilResponseTransport{},
+	}
+
+	return httpClient, mockHandler, mockServer
+}
+
 // NewMockAndHandler returns a Zappr Client that uses Github Token, Mockhandler, and Server. The client proxies
 // requests to the server and handlers can be registered on the mux to handle
 // requests. The caller must close the test server.
 func NewMockAndHandler() (Client, *test.MockHandler, *test.Server) {
 	return NewMockAndHandlerWithGithubToken("1234567890")
+}
+
+// NewMockAndHandlerNilResponse returns a Zappr Client that uses Github Token, Mockhandler, and Server.
+// The internal http client always returns a nil response object.
+func NewMockAndHandlerNilResponse() (Client, *test.MockHandler, *test.Server) {
+	httpClient, mockHandler, mockServer := newMockAndHandlerWithNilResponseTransport()
+	client := NewWithGithubToken("https://fake.zappr/", "1234567890", httpClient)
+
+	return client, mockHandler, mockServer
 }
 
 // NewMockAndHandlerWithGithubToken returns a Zappr Client that uses Github Token, Mockhandler, and Server. The client proxies
@@ -87,4 +108,15 @@ func (t *RewriteTransport) RoundTrip(req *http.Request) (*http.Response, error) 
 		return http.DefaultTransport.RoundTrip(req)
 	}
 	return t.Transport.RoundTrip(req)
+}
+
+// NilResponseTransport always returns a nil response object
+type NilResponseTransport struct {
+	Transport http.RoundTripper
+}
+
+// RoundTrip rewrites the request scheme to http and calls through to the
+// composed RoundTripper or if it is nil, to the http.DefaultTransport.
+func (t *NilResponseTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	return nil, errors.New("some error")
 }

